### ISSUE TITLE
DuoFern Room Thermostat - hvac_modes

### ISSUE THREAD

**Issue by hgsdfhjgs** [peribeir/homeassistant-rademacher#106]
Hi,
I moved to HA two month ago and I really enjoy this Integration because I have a lot of Rademacher devices.  It works really well. Big thank you for this work!!!! 

I have now a change request for “DuoFern Room Thermostat”.  I really like to see if my heating is running or not. Right now, the problem is the status “heat” is not visual in HA. There is no Icon and colour change. 

The reason is that the “hvac_modes” are wrong. It is currently used for “auto-mode” and “manual-mode” covert as Heat/Cool.
I think the correct “hvac_modes” would be “heat” or “off” because this is what a thermostat does.

The “auto-mode” and “manual-mode” is just a configuration inside the thermostat if the target temperature can be changed automatically or not.

I looked in the documentation for “Climate Entity” and I found “preset_modes”. Maybe we can use “auto-mode” and “manual-mode” as “preset_modes” just in case someone need to use it.

To sum up my suggestion are:

hvac_modes:
"relaisstatus":0 = HVACMode.OFF
"relaisstatus":1 = HVACMode.HEAT

preset_modes:
"Manuellbetrieb":0 = ECO(auto-mode)
"Manuellbetrieb":100 = BOOST (manual-mode)
 
Is it possible to change the behavior like that? Thanks in advance. 

Regards Chris

**Comment by peribeir** [comment]
Hi @hgsdfhjgs, I thought I had resolved this, but actually I forgot some things.
By correcting the HVACModes, I now need to know which command is used by Homepilot to turn on (HEAT) or off the thermostat.
Can you help here? Do you know the json payload that is sent to homepilot when you turn on of off? Can you use the browser developer tools on the web interface to find this out so that I can put it in the code?
Thanks

**Comment by hgsdfhjgs** [comment]
Hi @peribeir,
I'm not quite sure if I understand you correct but form the homepilot GUI there is no way to turn off or on the heat. I can only change the goal temperature and the thermostat will turn on or off the heat. 

"relaisstatus":0 = OFF
"relaisstatus":1 = ON

{"response":"get_device","device":{"description":"","deviceGroup":5,"did":1010058,"hasErrors":0,"iconSetInverted":0,"iconSet":{"k":"iconset28"},"messages":[],"name":"Heizung Wohnzimmer ","properties":{"closingContact":1,"dawn":3,"dusk":3,"motion":3,"rain":3,"smartphone":3,"smoke":3,"sun":3,"temperature":3,"time":2,"trigger":2,"warning":3,"wind":3},"statusValid":true,"statusesMap":{"Manuellbetrieb":0,"Position":170,"acttemperatur":236,"**relaisstatus":0**,"automaticvalue":190,"manualoverride":100},"visible":true,"deviceNumber":"32501812_A","uid":"73850a_A_1","voiceControlledBy":"","origin":"HomePilot","posMin":40,"posMax":400,"thresholds":[{"id":1,"value":230,"active":false},{"id":2,"value":210,"active":false},{"id":3,"value":190,"active":true},{"id":4,"value":150,"active":false}],"sceneConfig":["DIRECT_FUNCTION_CMD","CURR_TEMPERATURE_THRESH_CFG","AUTO_MODE_CFG"]}}

Do you need more informations? Thanks for your effort!

**Comment by Aeroschmelz** [comment]
It would be great if you could add the status of the thermostat which is given in the reading "relaisstatus". This would help to see if the heating is actually on or off and one could plot the state or do something else with it. With a Rest Sensor you can do readout the value like this:
`
- platform: rest
  name: "Room Thermostat Actuator"
  resource: '[http://Home-Pilot-IP]/devices/[DID]'
  value_template: >
    {%if (value_json["payload"]["device"]["capabilities"][14]["value"]) == 'true' %} 1  {% else %} 0 {% endif %}
  scan_interval: 60
`

**Comment by Grashuepfer18** [comment]
Hi @peribeir,

I have modified the climate.py file as follows. After these adjustments, the preset modes became available, and the heating system's state was displayed correctly.

Can you check it?

```
"""Platform for Rademacher Bridge."""
import asyncio
import logging

from homepilot.device import HomePilotDevice
from homepilot.manager import HomePilotManager
from homepilot.thermostat import HomePilotThermostat

from homeassistant.components.climate import ClimateEntity
from homeassistant.components.climate.const import (
    PRESET_BOOST,
    PRESET_ECO,
    PRESET_NONE,
    ClimateEntityFeature,
    HVACMode,
)
from homeassistant.const import CONF_EXCLUDE, UnitOfTemperature
from homeassistant.helpers.update_coordinator import DataUpdateCoordinator

from .const import DOMAIN
from .entity import HomePilotEntity

_LOGGER = logging.getLogger(__name__)


async def async_setup_entry(hass, config_entry, async_add_entities):
    """Setup of entities for sensor platform."""
    entry = hass.data[DOMAIN][config_entry.entry_id]
    manager: HomePilotManager = entry[0]
    coordinator: DataUpdateCoordinator = entry[1]
    exclude_devices: list[str] = entry[3][CONF_EXCLUDE]
    new_entities = []
    for did in manager.devices:
        if did not in exclude_devices:
            device: HomePilotDevice = manager.devices[did]
            if isinstance(device, HomePilotThermostat):
                _LOGGER.info("Found Thermostat for Device ID: %s", device.did)
                new_entities.append(
                    HomePilotClimateEntity(
                        coordinator,
                        device,
                        UnitOfTemperature.CELSIUS,
                    )
                )
    # If we have any new devices, add them
    if new_entities:
        async_add_entities(new_entities)


class HomePilotClimateEntity(HomePilotEntity, ClimateEntity):
    """This class represents all Sensors supported."""

    def __init__(
        self,
        coordinator,
        device: HomePilotThermostat,
        temperature_unit,
    ) -> None:
        super().__init__(
            coordinator,
            device,
            unique_id=f"{device.uid}",
            name=f"{device.name}",
            device_class=None,
        )
        self._attr_temperature_unit = temperature_unit
        self._attr_max_temp = device.max_target_temperature
        self._attr_min_temp = device.min_target_temperature
        self._attr_target_temperature_step = device.step_target_temperature
        self._attr_hvac_modes = (
            [HVACMode.HEAT, HVACMode.OFF]
        )
        self._attr_preset_modes = ([PRESET_ECO, PRESET_BOOST, PRESET_NONE])

    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
        device: HomePilotThermostat = self.coordinator.data[self.did]
        # if device.has_auto_mode:
        #     await device.async_set_auto_mode(hvac_mode == HVACMode.AUTO)
        await asyncio.sleep(5)
        await self.coordinator.async_request_refresh()

    async def async_set_temperature(self, **kwargs) -> None:
        device: HomePilotThermostat = self.coordinator.data[self.did]
        if device.can_set_target_temperature:
            await device.async_set_target_temperature(kwargs["temperature"])
            await asyncio.sleep(5)
            await self.coordinator.async_request_refresh()

    async def async_set_preset_mode(self, preset_mode: str) -> None:
        """Set the preset mode (ECO or BOOST)."""
        device: HomePilotThermostat = self.coordinator.data[self.did]
        
        if device.has_auto_mode:
            if preset_mode == PRESET_ECO:
                # ECO mode (manualoverride: 100)
                await device.async_set_auto_mode(True)
            elif preset_mode == PRESET_BOOST:
                # BOOST mode (manualoverride: 0)
                await device.async_set_auto_mode(False)
            else:
                _LOGGER.warning("Unknown preset mode: %s", preset_mode)
                return

            # Refresh data after setting mode
            async with asyncio.timeout(5):
                await self.coordinator.async_request_refresh()

    @property
    def current_temperature(self) -> float:
        device: HomePilotThermostat = self.coordinator.data[self.did]
        return device.temperature_value if device.has_temperature else None

    @property
    def target_temperature(self) -> float:
        device: HomePilotThermostat = self.coordinator.data[self.did]
        return (
            device.target_temperature_value if device.has_target_temperature else None
        )

    @property
    def preset_mode(self) -> str:
        """Return the current preset mode (ECO or BOOST)."""
        device: HomePilotThermostat = self.coordinator.data[self.did]
        if not device.has_auto_mode:
            return PRESET_NONE
        return PRESET_ECO if device.auto_mode_value else PRESET_BOOST

    @property
    def hvac_mode(self) -> str:
        device: HomePilotThermostat = self.coordinator.data[self.did]
        return (
            HVACMode.HEAT
            if device.has_relais_status and device.relais_status == 1
            else HVACMode.OFF
        )

    @property
    def supported_features(self) -> int:
        device: HomePilotThermostat = self.coordinator.data[self.did]
        features = 0
        
        # Unterstütze Zieltemperatur, wenn das Gerät sie setzen kann
        if device.can_set_target_temperature:
            features |= ClimateEntityFeature.TARGET_TEMPERATURE
            
        # Setze TURN_ON und TURN_OFF, da der HVAC-Modus auch diese implizit unterstützt
        if device.target_temperature_value and device.temperature_value:
            features |= ClimateEntityFeature.TURN_ON
            features |= ClimateEntityFeature.TURN_OFF
        
        if device.has_auto_mode:
            features |= ClimateEntityFeature.PRESET_MODE
    
        return features
```
